### PR TITLE
Don't run StringBuffer/Builder growth test in constrained environments

### PR DESCRIPTION
**Pull request by pshipton**
Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
This looks more like "disable this test on the IBM SDK".  Am I missing where it checks for resource constraints?

**Comment by pshipton** [comment]
> This looks more like "disable this test on the IBM SDK".

This is exactly what it does. The IBM test machines don't enough enough physically memory to run the tests. The JVM is created with -Xmx7g bu then the test times out trying to allocate the large objects, I'm guessing because the machine starts swapping.

**Comment by pshipton** [comment]
I didn't see any failures in the OpenJ9 test runs last night (although Windows didn't build at all).

**Comment by pshipton** [comment]
Updated the test to check the physical memory on the machine, and only run the test when there is 10GB or more.

File: test/functional/cmdLineTests/cmdLineTest_J9tests/src/TestStringBufferAndBuilderGrowth.java [hunk 1]
```diff
@@ -20,13 +20,25 @@
...
  * @file TestStringBufferAndBuilderGrowth.java
  * @brief Grows a StringBuffer and StringBuilder to Integer.MAX_VALUE
  */
 public class TestStringBufferAndBuilderGrowth {
 
 public static void main(String[] args) {
+	OperatingSystemMXBean opBean = (OperatingSystemMXBean)ManagementFactory.getOperatingSystemMXBean();
+	long physicalMemory = opBean.getTotalPhysicalMemorySize();
+	System.out.println("Machine has physical memory " + physicalMemory + " bytes or " + (physicalMemory >> 20) + " MB or " + (physicalMemory >> 30) + " GB");
+	long tenGig = 10L << 30;
```
Comment (keithc-ca) [inline test/functional/cmdLineTests/cmdLineTest_J9tests/src/TestStringBufferAndBuilderGrowth.java]: I wonder whether 8GB might be a more reasonable requirement, but I suppose we can try this and experiment later.

File: test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_Java8.xml [hunk 2]
```diff
@@ -550,7 +550,8 @@
   <command>$EXE$ -Xdump:none -Xmx7g -Djava.lang.stringBufferAndBuilder.growAggressively -cp $Q$$JARPATH$$Q$ TestStringBufferAndBuilderGrowth</command>
   <output regex="no" type="success">StringBuffer capacity=2147483647 StringBuilder capacity=2147483647</output>
   <output regex="no" type="success">Option too large</output>
- </test>
+  <output regex="no" type="success">Not enough resource to run test</output>
+</test>
```
Comment (keithc-ca) [inline test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_Java8.xml]: Wasn't the indentation of this correct before?

Comment (pshipton) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_Java8.xml]: Fixed.

File: test/functional/cmdLineTests/cmdLineTest_J9tests/src/TestStringBufferAndBuilderGrowth.java (hunk 1 above)
Comment (pshipton) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/src/TestStringBufferAndBuilderGrowth.java]: I know AIX doesn't work with 7616 MB, trying a 8000 MB limit.

Comment (keithc-ca) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/src/TestStringBufferAndBuilderGrowth.java]: I think 8000MB is a more reasonable boundary.

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,zlinux,win jdk11

**Comment by pshipton** [comment]
The internal testing didn't find any problems.

**Comment by pshipton** [comment]
It still fails on a machine with 8047 MB, I'll have to tweak it.

**Pull request merged**